service/src/integrationTest/kotlin/fi/espoo/evaka/messaging/MessageQueriesTest.kt

```diff
@@ -1194,6 +1198,249 @@ class MessageQueriesTest : PureJdbiTest(resetDbBeforeEach = true) {
         assertTrue(citizenAccounts.any { it.first == currentParentAccountId })
     }
 
+    companion object {
+        @JvmStatic
+        fun placementFilters(): Stream<Array<Any>> =
+            Stream.of(
+                arrayOf(listOf(MessagingCategory.MESSAGING_DAYCARE), 4),
+                arrayOf(listOf(MessagingCategory.MESSAGING_PRESCHOOL), 2),
+                arrayOf(listOf(MessagingCategory.MESSAGING_CLUB), 1),
+                arrayOf(
+                    listOf(MessagingCategory.MESSAGING_CLUB, MessagingCategory.MESSAGING_DAYCARE),
+                    5,
+                ),
+                arrayOf(
+                    listOf(
+                        MessagingCategory.MESSAGING_PRESCHOOL,
+                        MessagingCategory.MESSAGING_DAYCARE,
+                    ),
+                    6,
+                ),
+                arrayOf(listOf<MessagingCategory>(), 7),
+            )
+    }
+
+    @ParameterizedTest(name = "messagingCategoryList={0}, expectedRecipientCount={1}")
+    @MethodSource("placementFilters")
+    fun `getMessageAccountsForRecipients returns correct accounts for different placement type categories`(
+        messagingCategoryList: List<MessagingCategory>,
+        expectedRecipientCount: Int,
+    ) {
```

Comment:
Vähän nätimmin ja enemmän kotlin-natiiveja tyyppejä käyttäen onnistuu näin:

```suggestion
    private val placementFilters: List<Pair<List<MessagingCategory>, Int>> =
        listOf(
            listOf(MessagingCategory.MESSAGING_DAYCARE) to 4,
            listOf(MessagingCategory.MESSAGING_PRESCHOOL) to 2,
            listOf(MessagingCategory.MESSAGING_CLUB) to 1,
            listOf(MessagingCategory.MESSAGING_CLUB, MessagingCategory.MESSAGING_DAYCARE) to 5,
            listOf(MessagingCategory.MESSAGING_PRESCHOOL, MessagingCategory.MESSAGING_DAYCARE) to 6,
            listOf<MessagingCategory>() to 7,
        )

    @ParameterizedTest(name = "messagingCategoryList, expectedRecipientCount = {0}")
    @FieldSource("placementFilters")
    fun `getMessageAccountsForRecipients returns correct accounts for different placement type categories`(
        arg: Pair<List<MessagingCategory>, Int>
    ) {
        val (messagingCategoryList, expectedRecipientCount) = arg

```